cf-check/observables.c

```diff
@@ -0,0 +1,88 @@
+#include <platform.h>
+#include <observables.h>
+#include <logging.h>
+#include <definitions.h> // CF_BUFSIZE
+#include <known_dirs.h> // GetStateDir()
+#include <file_lib.h> // FILE_SEPARATOR
+#include <alloc.h> // xstrdup()
+#include <db_structs.h> // observable_strings
+
+void GetObservableNames(
+    const char *ts_key_path,
+    const char const** obnames)
+{
+    char buf[CF_BUFSIZE];
+    const char *filename;
+    const char const* temp[CF_OBSERVABLES] = {};
+    int i;
+
+    if (ts_key_path == NULL)
+    {
+        snprintf(buf, CF_BUFSIZE - 1, "%s%cts_key", GetStateDir(), FILE_SEPARATOR);
+        filename = buf;
+    }
+    else
+    {
+        filename = ts_key_path;
+    }
+
+    FILE *f = safe_fopen(filename, "r");
+    if (f == NULL)
+    {
+        for (i = 0; i < CF_OBSERVABLES; ++i)
+        {
+            if (i < observables_max)
+            {
+                temp[i] = observable_strings[i];
+            }
+            else
+            {
+                char buf[CF_MAXVARSIZE];
+                snprintf(buf, CF_MAXVARSIZE, "observable[%d]", i);
+                temp[i] = xstrdup(buf);
+            }
+        }
+    }
+    else
+    {
+        for (i = 0; i < CF_OBSERVABLES; ++i)
+        {
+            char line[CF_MAXVARSIZE];
+    
+            char name[CF_MAXVARSIZE], desc[CF_MAXVARSIZE];
+            char units[CF_MAXVARSIZE] = "unknown";
+            double expected_min = 0.0;
+            double expected_max = 100.0;
+            int consolidable = true;
+    
+            if (fgets(line, CF_MAXVARSIZE, f) == NULL)
+            {
+                Log(LOG_LEVEL_ERR, "Error trying to read ts_key from file '%s'. (fgets: %s)", filename, GetErrorStr());
+                break;
+            }
+    
+            int fields = sscanf(line, "%*d,%1023[^,],%1023[^,],%1023[^,],%lf,%lf,%d",
```

Comment:
I don't use most of the parsed data but figured I would leave it in as easy to compare to other code and capturing number of fields for that diagnostic.